science/fetcher.py

```diff
@@ -99,6 +135,7 @@ def _fetch_to_cache(
                     configured_client(url, headers).stream("GET", url) as response,
                     work.open("wb") as cache_fp,
                 ):
+                    response.raise_for_status()
```

Comment:
This missing `response.raise_for_status` and the one added below were likely bugs that may combine with the rest of this code to actually fix the OP issue. Discussion here: https://github.com/a-scie/lift/issues/116#issuecomment-2571731427

Even if not, I'd like to proceed with the change since retries seem like a good idea to me independently of #116.